Add transaction related changes and api docs (#24037)
ballerina-platform/ballerina-lang: stdlib/database/jdbc/src/main/ballerina/src/java.jdbc/Module.md

that is accessible via Java Database Connectivity (JDBC). 

**Prerequisite:** Add the JDBC driver corresponding to the database you are trying to interact with
as a native library dependency in your Ballerina project. Then, once you build the project with the `ballerina build`
command, you should be able to run the resultant JAR by executing the `java -jar` command.

E.g., The `Ballerina.toml` content.
Change the path to the JDBC driver appropriately.

```toml
[project]
org-name= "sample"
version= "0.1.0"

[platform]
target = "java8"

    [[platform.libraries]]
    artafactId = "h2"
    version = "1.4.200"
    path = "/path/to/com.h2database.h2-1.4.200.jar"
    groupId = "com.h2database"
    modules = ["samplemodule"]
``` 

Else, if you're trying to run a single BAL file, you can copy the JDBC driver into the `${BALLERINA_HOME}/bre/lib` directory and 
run the BAL file by executing the `ballerina run` command.

### Client
To access a database, you must first create a 
[jdbc:Client](https://ballerina.io/learn/api-docs/ballerina/api-docs/java.jdbc/clients/Client.html) object. 

> The examples for creating a JDBC client can be found below.

Reviewer: Shall we point to the BBE? Most of the other Module.md files we have pointed to the BBE

Author: For this case, I want to provide a link for the Client object definition and also providing the example here similar to BBE. And hence for this case, linking BBE seems to be not relevant.

## Module overview

This module provides the functionality required to access and manipulate data stored in any type of relational database 
that is accessible via Java Database Connectivity (JDBC). 

> **Prerequisite:** Please add the JDBC driver corresponding to the database you are trying to interact with, 

Reviewer 2: ```suggestion
**Prerequisite:** Add the JDBC driver corresponding to the database you are trying to interact with
```
The best practice is to avoid using "please" in technical documentation.

> as a native library dependency in your Ballerina project. Then, once you build the project with `ballerina build`

Reviewer 2: ```suggestion
as a native library dependency in your Ballerina project. Then, once you build the project with the `ballerina build`
```

> command, you should be able to run the resultant jar with, the `java -jar` command.

Reviewer 2: ```suggestion
command, you should be able to run the resultant JAR by executing the `java -jar` command.
```

> e.g. Ballerina.toml content.

Reviewer 2: ```suggestion
E.g., The `Ballerina.toml` content.
```

> Please change the path to the JDBC driver appropriately.

Reviewer 2: ```suggestion
Change the path to the JDBC driver appropriately.
```

```toml
[project]
org-name= "sample"
version= "0.1.0"

[platform]
target = "java8"

    [[platform.libraries]]
    artafactId = "h2"
    version = "1.4.200"
    path = "/path/to/com.h2database.h2-1.4.200.jar"
    groupId = "com.h2database"
    modules = ["samplemodule"]
``` 

> Or, if you're trying to run a single bal file, you can copy the JDBC driver into `${BALLERINA_HOME}/bre/lib` and 

Reviewer 2: ```suggestion
Else, if you're trying to run a single BAL file, you can copy the JDBC driver into the `${BALLERINA_HOME}/bre/lib` directory and 
```

> run the bal file with `ballerina run` command.

Reviewer 2: ```suggestion
run the BAL file by executing the `ballerina run` command.
```

### Client
To access a database, you must first create a 
[jdbc:Client](https://ballerina.io/learn/api-docs/ballerina/api-docs/java.jdbc/clients/Client.html) object. 
The examples for creating a JDBC client can be found below.

#### Creating a client

> This example shows different ways of creating the `jdbc:Client`. The client can be created by passing 

Reviewer 2: ```suggestion
This example shows the different ways of creating the `jdbc:Client`. The client can be created by passing 
```

> JDBC URL which is an mandatory property, and all other fields are optional. 

Reviewer 2: ```suggestion
the JDBC URL, which is a mandatory property and all other fields are optional. 
```

> The `dbClient1` receives only database URL, and the `dbClient2` receives user and password in addition to URL. 

Reviewer 2: ```suggestion
The `dbClient1` receives only the database URL and the `dbClient2` receives the username and password in addition to the URL. 
```

> If the properties are passed in the same order as it is defined in the `jdbc:Client` you can pass it 

Reviewer 2: ```suggestion
If the properties are passed in the same order as it is defined in the `jdbc:Client`, you can pass it 
```

without named params.

> The `dbClient3` uses the named params to pass all the attributes, and provides `options` property in the type of 

Reviewer 2: ```suggestion
The `dbClient3` uses the named params to pass all the attributes and provides the `options` property in the type of 
```

[jdbc:Options](https://ballerina.io/learn/api-docs/ballerina/api-docs/java.jdbc/records/Options.html) 
and also uses the unshared connection pool in the type of 
[sql:ConnectionPool](https://ballerina.io/learn/api-docs/ballerina/api-docs/sql/records/ConnectionPool.html). 

> Please refer [SQL Module](https://ballerina.io/learn/api-docs/ballerina/sql/index.html) to get more details 

Reviewer 2: ```suggestion
For more information about connection pooling, see [SQL Module](https://ballerina.io/learn/api-docs/ballerina/sql/index.html).
```

> about connection pooling.

Reviewer 2: ```suggestion

```

The `dbClient4` receives some custom properties within the 

> [jdbc:Options](https://ballerina.io/learn/api-docs/ballerina/api-docs/java.jdbc/records/Options.html), 

Reviewer 2: ```suggestion
[jdbc:Options](https://ballerina.io/learn/api-docs/ballerina/api-docs/java.jdbc/records/Options.html)
```

> and those properties will be used by the `datasourceName` defined. 

Reviewer 2: ```suggestion
and those properties will be used by the defined `datasourceName`. 
```

> As per the provided example, the datasource `org.h2.jdbcx.JdbcDataSource` will be configured with `loginTimeout` 

Reviewer 2: ```suggestion
As per the provided example, the `org.h2.jdbcx.JdbcDataSource` datasource  will be configured with a `loginTimeout` 
```

of `2000` milli seconds.

```ballerina
jdbc:Client dbClient1 = new ("jdbc:h2:~/path/to/database");
jdbc:Client dbClient2 = new ("jdbc:h2:~/path/to/database", "root", "root");
jdbc:Client dbClient3 = new (url =  "jdbc:h2:~/path/to/database",
                             user = "root", password = "root",
                             options = {
                                 datasourceName: "org.h2.jdbcx.JdbcDataSource"
                             },
                             connectionPool = {
                                 maxOpenConnections: 5
                             });
jdbc:Client dbClient4 = new (url =  "jdbc:h2:~/path/to/database", 
                             user = "root", password = "root",
                             options = {
                                datasourceName: "org.h2.jdbcx.JdbcDataSource", 
                                properties: {"loginTimeout": "2000"}
                             });                          
```

> You can find more details about each property in 

Reviewer 2: ```suggestion
You can find more details about each property in the
```

[jdbc:Client](https://ballerina.io/learn/api-docs/ballerina/api-docs/java.jdbc/clients/Client.html) constructor. 

The [jdbc:Client](https://ballerina.io/learn/api-docs/ballerina/api-docs/java.jdbc/clients/Client.html) references 
[sql:Client](https://ballerina.io/learn/api-docs/ballerina/api-docs/sql/clients/Client.html) and 

> hence all operations defined by the `sql:Client` will be supported by `jdbc:Client` as well. 

Reviewer 2: ```suggestion
all the operations defined by the `sql:Client` will be supported by the `jdbc:Client` as well. 
```

> Please refer [SQL Module](https://ballerina.io/learn/api-docs/ballerina/sql/index.html) for more information on 

Reviewer 2: ```suggestion
For more information on all the operations supported by the `jdbc:Client`, which include the below, see the
[SQL Module](https://ballerina.io/learn/api-docs/ballerina/sql/index.html).
```

> all operations supported by `jdbc:Client` which includes below. 

Reviewer 2: ```suggestion

```
